tools/ja-rule/ja-rule-controller.cpp

```diff
@@ -208,32 +183,33 @@ class InputHandler {
 
     uint8_t param_data = identify_on;
     RDMSetRequest *request = new RDMSetRequest(
-        m_widget_uid, m_selected_uid, 0, 0, 0, ola::rdm::PID_IDENTIFY_DEVICE,
+        m_widget->GetUID(), m_selected_uid, 0, 0, 0,
+        ola::rdm::PID_IDENTIFY_DEVICE,
         &param_data, sizeof(param_data));
-    widget->SendRDMRequest(request, NULL);
+    m_port->SendRDMRequest(request, NULL);
   }
 
   void RunDiscovery(bool incremental) {
-    JaRuleWidget *widget = m_widget_manager->GetWidget();
-    if (!widget) {
+    if (!m_widget) {
       return;
     }
     if (incremental) {
-      widget->RunIncrementalDiscovery(
-          NewSingleCallback(this, &InputHandler::DiscoveryComplete));
+      m_port->RunIncrementalDiscovery(
+          NewSingleCallback(this, &Controller::DiscoveryComplete));
     } else {
-      widget->RunFullDiscovery(
-          NewSingleCallback(this, &InputHandler::DiscoveryComplete));
+      m_port->RunFullDiscovery(
+          NewSingleCallback(this, &Controller::DiscoveryComplete));
     }
   }
 
   void ResetDevice() {
-    JaRuleWidget *widget = m_widget_manager->GetWidget();
-    if (!widget) {
+    /*
```

Comment:
It is, but I need to think about how to add it. It can either be a the device level or at the port level.